src/commands/service/create.ts

```diff
@@ -10,328 +10,144 @@ import {
   addNewService as addNewServiceToBedrockFile,
   YAML_NAME as BedrockFileName
 } from "../../lib/bedrockYaml";
+import { build as buildCmd, exit as exitCmd } from "../../lib/commandBuilder";
 import {
   addNewServiceToMaintainersFile,
   generateDockerfile,
   generateGitIgnoreFile,
   generateStarterAzurePipelinesYaml
 } from "../../lib/fileutils";
 import { checkoutCommitPushCreatePRLink } from "../../lib/gitutils";
+import { isPortNumber } from "../../lib/validator";
 import { logger } from "../../logger";
 import { IBedrockFileInfo, IHelmConfig, IUser } from "../../types";
+import decorator from "./create.decorator.json";
+
+export interface ICommandOptions {
+  displayName: string;
+  gitPush: boolean;
+  helmChartChart: string;
+  helmChartRepository: string;
+  helmConfigBranch: string;
+  helmConfigGit: string;
+  helmConfigPath: string;
+  k8sBackend: string;
+  maintainerEmail: string;
+  maintainerName: string;
+  middlewares: string;
+  packagesDir: string;
+  pathPrefix: string;
+  pathPrefixMajorVersion: string;
+  k8sBackendPort: string;
+}
+
+export interface ICommandValues extends ICommandOptions {
+  k8sPort: number;
+  middlewaresArray: string[];
+  variableGroups: string[];
+}
+
+export const fetchValues = (opts: ICommandOptions) => {
+  if (!isPortNumber(opts.k8sBackendPort)) {
+    throw new Error("value for --k8s-service-port is not a value port number");
+  }
+
+  let variableGroups: string[] = [];
+
+  try {
+    const bedrock = Bedrock();
+    variableGroups = bedrock.variableGroups || [];
+  } catch (_) {
+    // NO-OP
+  }
+
+  let middlewaresArray: string[] = [];
+  if (opts.middlewares && opts.middlewares.trim()) {
+    middlewaresArray = opts.middlewares.split(",").map(str => str.trim());
+  }
+
+  const values: ICommandValues = {
+    displayName: opts.displayName,
+    gitPush: opts.gitPush,
+    helmChartChart: opts.helmChartChart,
+    helmChartRepository: opts.helmChartRepository,
+    helmConfigBranch: opts.helmConfigBranch,
+    helmConfigGit: opts.helmConfigGit,
+    helmConfigPath: opts.helmConfigPath,
+    k8sBackend: opts.k8sBackend,
+    k8sBackendPort: opts.k8sBackendPort,
+    k8sPort: parseInt(opts.k8sBackendPort, 10),
+    maintainerEmail: opts.maintainerEmail,
+    maintainerName: opts.maintainerName,
+    middlewares: opts.middlewares,
+    middlewaresArray,
+    packagesDir: opts.packagesDir,
+    pathPrefix: opts.pathPrefix,
+    pathPrefixMajorVersion: opts.pathPrefixMajorVersion,
+    variableGroups
+  };
+
+  // values need not be validated (that's do not need
```

Comment:
for clarity, the reason we typecheck the the values from `opts` is because `opts` has a type of `{[key: string]: any}`. So when your create your `ICommandValues`, we are implicitly casting those values to whatever the key in `ICommandValues` points to.